TestFoundation/TestNSURLResponse.swift

```diff
@@ -60,4 +61,208 @@ class TestNSURLResponse : XCTestCase {
         let res2 = NSURLResponse(URL: url, MIMEType: nil, expectedContentLength: 0, textEncodingName: nil)
         XCTAssertNil(res2.textEncodingName)
     }
-}
\ No newline at end of file
+    
+    func test_suggestedFilename() {
+        let url = NSURL(string: "a/test/path")!
+        let res = NSURLResponse(URL: url, MIMEType: "txt", expectedContentLength: 0, textEncodingName: nil)
+        XCTAssertNil(res.suggestedFilename)
+    }
+}
+
+
+class TestNSHTTPURLResponse : XCTestCase {
+    static var allTests: [(String, TestNSHTTPURLResponse -> () throws -> Void)] {
+        return [
+                   ("test_URL_and_status_1", test_URL_and_status_1),
+                   ("test_URL_and_status_2", test_URL_and_status_2),
+
+                   ("test_headerFields_1", test_headerFields_1),
+                   ("test_headerFields_2", test_headerFields_2),
+                   ("test_headerFields_3", test_headerFields_3),
+                   
+                   ("test_contentLength_available_1", test_contentLength_available_1),
+                   ("test_contentLength_available_2", test_contentLength_available_2),
+                   ("test_contentLength_available_3", test_contentLength_available_3),
+                   ("test_contentLength_available_4", test_contentLength_available_4),
+                   ("test_contentLength_notAvailable_1", test_contentLength_notAvailable_1),
+                   ("test_contentLength_notAvailable_2", test_contentLength_notAvailable_2),
+                   ("test_contentLength_notAvailable_3", test_contentLength_notAvailable_3),
+                   ("test_contentLength_notAvailable_4", test_contentLength_notAvailable_4),
+                   ("test_contentLength_notAvailable_5", test_contentLength_notAvailable_5),
+                   
+                   ("test_suggestedFilename_notAvailable_1", test_suggestedFilename_notAvailable_1),
+                   ("test_suggestedFilename_notAvailable_2", test_suggestedFilename_notAvailable_2),
+                   ("test_suggestedFilename_1", test_suggestedFilename_1),
+                   ("test_suggestedFilename_2", test_suggestedFilename_2),
+                   ("test_suggestedFilename_3", test_suggestedFilename_3),
+                   ("test_suggestedFilename_4", test_suggestedFilename_4),
+                   ("test_suggestedFilename_onlyTheLastPathComponent_1", test_suggestedFilename_onlyTheLastPathComponent_1),
+                   ("test_suggestedFilename_onlyTheLastPathComponent_2", test_suggestedFilename_onlyTheLastPathComponent_2),
+                   
+                   ("test_MIMETypeAndCharacterEncoding_1", test_MIMETypeAndCharacterEncoding_1),
+                   ("test_MIMETypeAndCharacterEncoding_2", test_MIMETypeAndCharacterEncoding_2),
+                   ("test_MIMETypeAndCharacterEncoding_3", test_MIMETypeAndCharacterEncoding_3),
```

Comment:
I wish the swift-ci could run these the objc foundation. looks pretty comprehensive. I will have to run these tests manually later to fully review this.